tests/trainer/test_dataloaders.py

```diff
@@ -1032,23 +1032,77 @@ def test_dataloaders_load_only_once_no_sanity_check(tmpdir):
 @mock.patch.dict(os.environ, {"PL_DEV_DEBUG": "1"})
 def test_dataloaders_load_every_epoch(tmpdir):
 
-    model = EvalModelTemplate()
+    model = BoringModel()
 
-    # logger file to get meta
     trainer = Trainer(
         default_root_dir=tmpdir,
         limit_train_batches=0.3,
         limit_val_batches=0.3,
-        reload_dataloaders_every_epoch=True,
+        reload_dataloaders_every_n_epochs=True,
         max_epochs=3,
     )
-    result = trainer.fit(model)
+    trainer.fit(model)
+    trainer.test()
 
+    # verify the sequence
+    calls = trainer.dev_debugger.dataloader_sequence_calls
+    expected_sequence = [
+        'val_dataloader',
+        'train_dataloader',
+        'val_dataloader',
+        'train_dataloader',
+        'val_dataloader',
+        'train_dataloader',
+        'val_dataloader',
+        'test_dataloader'
+    ]
+    for call, expected in zip(calls, expected_sequence):
+        assert call['name'] == expected
+
+
+@mock.patch.dict(os.environ, {"PL_DEV_DEBUG": "1"})
+def test_dataloaders_load_every_n_epochs(tmpdir):
+
+    model = BoringModel()
+
+    trainer = Trainer(
+        default_root_dir=tmpdir,
+        limit_train_batches=0.3,
+        limit_val_batches=0.3,
+        reload_dataloaders_every_n_epochs=2,
+        max_epochs=3,
+    )
+    trainer.fit(model)
     trainer.test()
 
-    assert len(trainer.dev_debugger.val_dataloader_calls) == 4
-    assert len(trainer.dev_debugger.train_dataloader_calls) == 3
-    assert len(trainer.dev_debugger.test_dataloader_calls) == 1
+    # verify the sequence
+    calls = trainer.dev_debugger.dataloader_sequence_calls
+    expected_sequence = [
+        'val_dataloader',
+        'train_dataloader',
+        'val_dataloader',
+        'train_dataloader',
+        'val_dataloader',
+        'test_dataloader'
+    ]
+    for call, expected in zip(calls, expected_sequence):
+        assert call['name'] == expected
+
+
+@mock.patch.dict(os.environ, {"PL_DEV_DEBUG": "1"})
+def test_dataloaders_load_every_epoch_depreciated(tmpdir):
```

Comment:
```suggestion
def test_dataloaders_load_every_epoch_deprecated(tmpdir):
```

also, I believe we have separate test files for deprecated tests, see for example https://github.com/PyTorchLightning/pytorch-lightning/blob/master/tests/deprecated_api/test_remove_1-3.py

Reply:
got it. so I'll remove this test from here and add it to a _test_remove_1-4.py_